src/lib.rs

```diff
@@ -72,8 +72,26 @@ impl Clipboard {
     }
 }
 
+impl Clipboard {
+    pub fn read_primary(&self) -> Option<Result<String, Box<dyn Error>>> {
+        self.raw.read_primary()
+    }
+
+    pub fn write_primary(&mut self, contents: String) -> Option<Result<(), Box<dyn Error>>> {
+        self.raw.write_primary(contents)
+    }
+}
+
 pub trait ClipboardProvider {
     fn read(&self) -> Result<String, Box<dyn Error>>;
 
     fn write(&mut self, contents: String) -> Result<(), Box<dyn Error>>;
+
+    fn read_primary(&self) -> Option<Result<String, Box<dyn Error>>> {
+        None
+    }
+
+    fn write_primary(&mut self, _contents: String) -> Option<Result<(), Box<dyn Error>>> {
+        None
+    }
```

Comment:
Can we make these available only on Linux?

Reply:
Both me and @MoSal created a fix for this, I choose to expose this for unix only, but that is really not very nice to have in the api, since that leaks up all the way through iced to applications trying to use this. So, even if it is more technically correct to only expose this for unix systems, it is much more clean and user friendly with this implementation that have a default implementation that returns None on systems without primary.

Reply:
Fair enough!